### DISCUSSION TITLE
ComboBoxItem勾选联动一个datagrid的某一行失效（变灰）该如何实现？

### DISCUSSION THREAD

**Discussion by Monoideism** [BYJRK/DotNet-Discussions#8] [WPF 开发]
如题，现有一个 `ComboBox`，假设其包含 3 个 `ComboBoxItem`，已经设置好
这个 `ComboBox` 下方还有个 `DataGrid`, 分为 3 列。大致组织形式为：

```xml
<DataGrid Grid.Row="1" Grid.RowSpan="2" Name="DG1">
    <DataGrid.Columns>
        <DataGridTextColumn Header="_Parameter1" Width="auto"  Binding="{Binding Option}" >
        </DataGridTextColumn>
        <DataGridTextColumn Header="_Value" Width="auto" Binding="{Binding Pvalue}" >
        </DataGridTextColumn>
        <DataGridTextColumn Header="Fitness" Width="auto"  Binding="{Binding Unit}"  >
        </DataGridTextColumn>
    </DataGrid.Columns>
 </DataGrid>
```

在 xaml.cs 中 ，3 次 OP 实例化写列，生成了 3 行

```csharp
public partial class MainWindow : Window
{      
    public ObservableCollection<Properties> OP { get; set; }
    public MainWindow ()
    {
        InitializeComponent ();
        OP = new ()
        {
            new Properties () { Option = "Shortest wavelength", Pvalue = 550, Unit = "[nm]" },
            new Properties () { Option = "Longest wavelength", Pvalue = 3800, Unit = "[nm]" },
            new Properties () { Option = "Number of wavelengths", Pvalue = 6, Unit = "[]" }
        };
        DG1.ItemsSource = SpectralProperties;
    } 
}

// 还有个 Properties.cs 存放 
public string option { get; set; }  
public string value { get; set; }
public string unit { get; set; }
```

现在想写一个简单逻辑：

勾选 `ComboBox` 某一项，触发 `DataGrid` 里某行失效（类似灰度改变）（每次 `ObservableCollection<Properties> OP` 实例化是一次写 3 列，1 行）。该怎么写逻辑？

通过写类似 `if (combbox.selectedItem==1) { /* 里面逻辑不会写了 */ }` 这种，发现好像不产生什么变化。求指教！

**Comment by BYJRK** [comment]
你可以让 `Properites` 里面新增一个 bool 属性，表示这一行是否失效。然后在触发 `combobox.SelectionChanged` 时，根据情况调整你的 OP 中对应 item 的这个 bool 属性。然后在前台写一下 `DataGrid.RowStyle`，里面添加一个 `DataTrigger`，根据 bool 值的情况修改当前行的 `Background` 以及 `IsEnabled` 之类

**Reply by Monoideism** [reply]
“DataGrid.RowStyle，里面添加一个 DataTrigger” 是我的知识盲区了，我先学下。谢谢
现在我这样做：
private void Combobox1Input_SelectionChanged(object sender, SelectionChangedEventArgs e)
        {
            ComboBox cb= sender as ComboBox;
            ComboBoxItem cbi = cb.SelectionBoxItem as ComboBoxItem;
            if (cbi != null)
            {
                if (cbi.Content.ToString() == "AAA")
                    button1.Background = Brushes.Green;
                else if (cbi.Content.ToString() == "BBB")
                    DGridOptParameter.RowDetailsVisibilityMode = DataGridRowDetailsVisibilityMode.Collapsed;
                else if (cbi.Content.ToString() == "CCC")
                    button1.Background = Brushes.Red;
            }
            
            button1颜色没有任何变化，不知道哪里出问题传不过去

**Reply by BYJRK** [reply]
打断点进去看看cbi.Content.ToString()是什么。照理说如果你能拿到button1的引用，然后修改了Background，那不会不起作用的

**Reply by Monoideism** [reply]
“表示这一行是否失效” 这一行我不知道怎么拿到。。。我的这3行是用一列一列分3次生成  获得的。

**Reply by Monoideism** [reply]
> 打断点进去看看cbi.Content.ToString()是什么。照理说如果你能拿到button1的引用，然后修改了Background，那不会不起作用的

太菜了。。。感谢。

**Reply by Monoideism** [reply]
> 打断点进去看看cbi.Content.ToString()是什么。照理说如果你能拿到button1的引用，然后修改了Background，那不会不起作用的

是Null...
